File: 2024/10 - October/23. Find the Sum of Last N nodes of the Linked List/Solution.java
class Solution 
{
    public int sumOfLastN_Nodes(Node head, int n) 
    {
        if(head == null) return 0;
        
        Node temp = head;
        int length = 0;
        
        while(temp != null)
        {
            length++;
            temp = temp.next;
        }
        
        int startPos = length - n;
        temp = head;
        
        while(startPos > 0)
        {
            temp = temp.next;
            startPos--;
        }
        
        int res = 0;
        while(temp != null)
        {
            res += temp.data;
            temp = temp.next;
        }
        
        return res;
    }
}

File: 2024/10 - October/23. Find the Sum of Last N nodes of the Linked List/Solution.cpp
class Solution 
{
  public:
    int sumOfLastN_Nodes(struct Node* head, int n) 
    {
       Node* temp = head;
        int length = 0;
        while (temp != NULL) 
        {
            length++;
            temp = temp->next;
        }
        
        int startPos = length - n;
        temp = head;
        
        while (startPos > 0) 
        {
            temp = temp->next;
            startPos--;
        }
        
        int sum = 0;
        while (temp != NULL) 
        {
            sum += temp->data;
            temp = temp->next;
        }
        
        return sum;
    }
};
